src/fingerprint_bond.cpp

```
@@ -0,0 +1,917 @@
+to publish this information. Any opinions, findings and conclusions or
+recommendations expressed in this material are those of the author(s) and
+do not necessarily reflect the views of the United States Army.​”
+
+DISTRIBUTION A. Approved for public release; distribution unlimited. OPSEC#4918
+ */
+
+#include "fingerprint_bond.h"
+
+
+using namespace LAMMPS_NS;
+
+Fingerprint_bond::Fingerprint_bond(PairRANN *pair) : Fingerprint(pair)
+{
+	n_body_type = 3;
+	dr = 0;
+	re = 0;
+	rc = 0;
+	alpha_k = new double [1];
+	alpha_k[0] = -1;
+	k = 0;
+	m = 0;
+	id = -1;
+	style = "bond";
+	atomtypes = new int [n_body_type];
+	empty = true;
+	pair->allscreen = false;
+}
+
+Fingerprint_bond::~Fingerprint_bond(){
+	delete [] alpha_k;
+	delete [] atomtypes;
+	delete [] expcuttable;
+	delete [] dfctable;
+	for (int i=0;i<(m*(m+1))>>1;i++){
+		delete [] coeff[i];
+		delete [] coeffx[i];
+		delete [] coeffy[i];
+		delete [] coeffz[i];
+		delete [] Mf[i];
+	}
+	delete [] coeff;
+	delete [] coeffx;
+	delete [] coeffy;
+	delete [] coeffz;
+	delete [] Mf;
+	delete [] rinvsqrttable;
+}
+
+bool Fingerprint_bond::parse_values(char * constant, char * line1){
+	char **words=new char *[MAXLINE];
+	int nwords,l;
+	nwords=0;
+	words[nwords++] = strtok(line1,": ,\t\n");
+	while ((words[nwords++] = strtok(NULL,": ,\t\n"))) continue;
+	nwords -= 1;
+	if (strcmp(constant,"re")==0){
+		re = strtod(words[0],NULL);
+	}
+	else if (strcmp(constant,"rc")==0){
+		rc = strtod(words[0],NULL);
+	}
+	else if (strcmp(constant,"alphak")==0){
+		delete [] alpha_k;
+		alpha_k = new double [nwords];
+		for (l=0;l<nwords;l++){
+			alpha_k[l]=strtod(words[l],NULL);
+		}
+	}
+	else if (strcmp(constant,"dr")==0){
+		dr = strtod(words[0],NULL);
+	}
+	else if (strcmp(constant,"k")==0){
+		k = strtol(words[0],NULL,10);
+	}
+	else if (strcmp(constant,"m")==0){
+		m = strtol(words[0],NULL,10);
+	}
+	else pair->errorf("Undefined value for bond power");
```

This kind of error condition should either throw an exception or directly call `Error::one()` or `Error::all()`, you should be able to access those via `pair->lmp->error->`.

---

I have tried to make all the fingerprints and activations rely only on things defined by pair_rann. This is because I want to use the same code with minimal changes in my calibration routines outside of LAMMPS. For that reason, I put a wrapper around the error function, created a struct with the NeighborList information, etc. I've modified the error function now to pass the correct line and file to LAMMPS Error::one().